zha/application/platforms/binary_sensor/__init__.py

```diff
@@ -75,6 +75,7 @@ def __init__(
         self._cluster_handler = cluster_handlers[0]
         super().__init__(cluster_handlers, endpoint, device, **kwargs)
         self._state: bool = self.is_on
+        self.recompute_capabilities()
```

Comment:
This method call is a bit weird in ZHA in general. `self.recompute_capabilities()` _should_ be called as part of creating the instance but we don't have a way to do that in the base class.

Reply:
Yeah, I noticed 😄